hr_attendance_calendar/models/hr_employee.py

```diff
@@ -48,20 +50,23 @@ class HrEmployee(models.Model):
 
     @api.multi
     @api.depends('attendance_days_ids.extra_hours', 'annual_balance')
-    def _compute_extra_hours(self):
-        for employee in self:
-
-            start_year = datetime.date.today().replace(month=1, day=1)
-            start_year = fields.Date.to_string(start_year)
-            yesterday = datetime.date.today() - datetime.timedelta(days=1)
-            yesterday = fields.Date.to_string(yesterday)
+    def _compute_extra_hours(self, start_date=None, end_date=None,
+                             from_start_of_employment=True):
+        if not start_date:
+            start_date = datetime.date.today().replace(month=1, day=1)
+        start_date_str = fields.Date.to_string(start_date)
+        if not end_date:
+            end_date = datetime.date.today() - datetime.timedelta(days=1)
+        end_date_str = fields.Date.to_string(end_date)
 
+        for employee in self:
             attendance_day_ids = employee.attendance_days_ids.filtered(
-                lambda r: start_year <= r.date <= yesterday)
+                lambda r: start_date_str <= r.date <= end_date_str)
 
             extra_hours_sum = sum(attendance_day_ids.mapped('extra_hours'))
-
-            employee.extra_hours = extra_hours_sum + employee.annual_balance
+            employee.extra_hours = extra_hours_sum
+            if from_start_of_employment:
+                employee.extra_hours += employee.annual_balance
```

Comment:
Assign the result in a temp var and set employee.extra_hours only once at the end of the loop to avoid writing twice in the object (which could unnecessarily trigger twice the compute methods on fields depending on extra_hours)